cEP-0020.md

```diff
@@ -0,0 +1,355 @@
+| Title    | Newcomer metrics and Gamification system               |
+| Authors  | Shrikrishna Singh <mailto:krishnasingh.ss30@gmail.com> |
+| Status   | Proposed                                               |
+| Type     | Process                                                |
+
+## Abstract
+
+This cEP describes how a gamification system will automate the process
+of becoming a `Developer` from a `Newcomer`, as a part of this
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5892040252981248).
+
+For the implementation of this process, we will incorporate a gamification
+system which will allow us to track newcomer's progress and assign some
+points to each of the activity they do. We will also have different levels
+and give them badges on the basis of points earned and performed activities
+by the newcomers.
+
+So, every newcomer at coala will have to complete the processes involved in
+the gamification system in order to become a developer at coala or to get the
+**coala Developer** badge.
+
+## Background
+
+### Newcomer Experience
+
+**The problems we have:**
+
+- How to attract more newcomers
+  - How to cope with the increasing amounts of newcomers
+- How to teach and help newcomers to learn faster so
+  we can keep working on coala at a high pace.
+- How to improve newcomer -> dev/maintainer ratio
+- How to increase amount of feedback we get
+  - How to use feedback to improve coala and processes
+
+## Why Gamification?
+
+As coala tries to be a welcoming organization for newcomers, it gives them
+clear pathways and as much direct assistance as possible. However, it is
+already known that newcomers face many barriers while attempting to contribute
+to the open source community for the first time. Some barriers they face include
+orientation issues that can potentially demotivate newcomers from placing
+their first contribution. On the other hand, gamification is widely used to
+engage and motivate people to accomplish tasks and improve their performance.
+Therefore, the goal of this project is to use gamification to orient and
+motivate newcomers to overcome onboarding barriers to contribute and engage
+with coala and its community.
+
+## Gamification Elements
+
+There will be three game design elements to help newcomers: Points, Levels
+and Badges. For each of these elements, we will have a set of rules to
+describe the operation of the gamified environment. i.e.:
+
+### Points and Rules
+
+Let’s start by defining some rules for earning points. Points are a
+simple tool to reward behavior, and they provide an excellent base upon which to
+build a gamified design. The first step is to define the activities that we want
+newcomers to take. i.e.:
+
+- Created Issues by a Newcomer
+- Created Pull Requests by a Newcomer
+- Number of Gitmate errors on the PR
+- CI status of the PR
+- Number of reviews done by a Newcomer
+- Number of comments in one review by a Newcomer
```

Comment:
Some people can just review for the sake of points - `As @prnvdixit said in his review........`. Maybe add a system to allow PR owner to give points to the reviewer?

Reply:
Yeah, I also had an idea for that is to also check the reactions on the review comments, E.g.: if its thumbs up, we will assign him the points and vice versa, But this thing is taking into consideration in the very vague way in the meta-review project and as John said [here](https://github.com/coala/cEPs/pull/131#discussion_r187619878) 
>it will be providing data to the community data model

Reply:
Makes sense then :smile: